providers/s3/s3.go

```diff
@@ -439,10 +439,17 @@ func (b *Bucket) getRange(ctx context.Context, name string, off, length int64) (
 	}
 
 	// StatObject to see if the object exists and we have permissions to read it
-	if _, err := b.client.StatObject(ctx, b.name, name, *opts); err != nil {
+	var stat minio.ObjectInfo
+	if stat, err = b.client.StatObject(ctx, b.name, name, *opts); err != nil {
 		return nil, err
 	}
-	return b.client.GetObject(ctx, b.name, name, *opts)
+
+	o, err := b.client.GetObject(ctx, b.name, name, *opts)
+	if err != nil {
+		return o, err
+	}
+
+	return objstore.ObjectSizerReadCloser{ReadCloser: o, Size: stat.Size}, nil
```

Comment:
references:

size is taken from `Content-Length` header
https://github.com/minio/minio-go/blob/master/utils.go#L264
https://docs.aws.amazon.com/AmazonS3/latest/API/API_HeadObject.html
```
HeadObject returns only the metadata for an object. If the Range is satisfiable, only the ContentLength is affected in the response. If the Range is not satisfiable, S3 returns a 416 - Requested Range Not Satisfiable error.
```

Reply:
Isnt this adding an additional request to the GetRange call?

Reply:
minio-go [saves the objectInfo](https://github.com/minio/minio-go/blob/master/api-get-object.go#L326-L335) when we get the object. [calling Stat()](https://github.com/minio/minio-go/blob/master/api-get-object.go#L428) after a successful `GetObject()` call won't trigger the `HEAD` request i think.

this is very likely not going to cause a regression since this code executes only when some asks for `Size()`.
but let me know if you notice any problems :)